yarp/yarp_compiler.c

```diff
@@ -1335,6 +1335,45 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
 
           return;
       }
+      case YP_NODE_INTERPOLATED_X_STRING_NODE: {
+          yp_interpolated_x_string_node_t *interp_x_string_node = (yp_interpolated_x_string_node_t *) node;
+          size_t parts_size = interp_x_string_node->parts.size;
+
+          ADD_INSN(ret, &dummy_line_node, putself);
+
+          if (parts_size > 0) {
+              for (size_t index = 0; index < parts_size; index++) {
+                  yp_node_t *part = interp_x_string_node->parts.nodes[index];
+
+                  switch (part->type) {
+                    case YP_NODE_STRING_NODE: {
+                        yp_x_string_node_t *x_string_node = (yp_x_string_node_t *) part;
```

Comment:
This is the wrong type here, since you're checking it's a `STRING_NODE` it should be `yp_string_node_t*`. From the other interpolated nodes, it looks like it matches those loops exactly, could you extract this out? Looks like it could be a function that takes a `yp_node_list_t*`.